rxandroidble/src/main/java/com/polidea/rxandroidble2/exceptions/BleDisconnectedException.java

```diff
@@ -12,26 +14,38 @@
  */
 public class BleDisconnectedException extends BleException {
 
+    /**
+     * Set when the state is not available, for example when the adapter has been switched off.
+     */
+    public static final int UNKNOWN_STATE = -1;
+
+    public static class BleAdapterDisabledException extends BleException {
+        // Disconnection related to disabled Bluetooth adapter
+    }
+
     @SuppressWarnings("WeakerAccess")
     @NonNull
     public final String bluetoothDeviceAddress;
+    public final int state;
 
-    @Deprecated
-    public BleDisconnectedException() {
-        bluetoothDeviceAddress = "";
+    public static BleDisconnectedException adapterDisabled(String macAddress) {
+        return new BleDisconnectedException(new BleDisconnectedException.BleAdapterDisabledException(), macAddress, UNKNOWN_STATE);
     }
 
-    public BleDisconnectedException(Throwable throwable, @NonNull String bluetoothDeviceAddress) {
-        super(createMessage(bluetoothDeviceAddress), throwable);
+    public BleDisconnectedException(Throwable throwable, @NonNull String bluetoothDeviceAddress, int state) {
```

Comment:
status
Plus this changes the public API :/

Reply:
Yes, should be status. Do you think that we should keep exception constructors as a part of the public API? If so, I can create a new one and deprecate the old one.

Reply:
We could deprecate the current constructors and add a `LIBRARY_GROUP` annotation to the new one